master/buildbot/test/integration/test_integration_secrets_with_vault.py

```diff
@@ -29,36 +29,32 @@
 
 
 class SecretsConfig(RunMasterBase):
+    def runOnDocker(self, args, **kwargs):
+        try:
+            subprocess.check_call(args, **kwargs)
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            raise SkipTest("Vault integration needs docker environment to be setup")
 
     def setUp(self):
-        try:
-            rv = subprocess.call(['docker', 'pull', 'vault'])
-            if rv != 0:
-                raise FileNotFoundError('docker')
-        except FileNotFoundError as e:
-            raise SkipTest(
-                "Vault integration need docker environment to be setup") from e
-
-        rv = subprocess.call(['docker', 'run', '-d',
-                              '-e', 'SKIP_SETCAP=yes',
-                              '-e', 'VAULT_DEV_ROOT_TOKEN_ID=my_vaulttoken',
-                              '-e', 'VAULT_TOKEN=my_vaulttoken',
-                              '--name=vault_for_buildbot',
-                              '-p', '8200:8200', 'vault'])
-        self.assertEqual(rv, 0)
+        self.runOnDocker(['docker', 'pull', 'vault'])
+
+        self.runOnDocker(['docker', 'run', '-d',
+                          '-e', 'SKIP_SETCAP=yes',
+                          '-e', 'VAULT_DEV_ROOT_TOKEN_ID=my_vaulttoken',
+                          '-e', 'VAULT_TOKEN=my_vaulttoken',
+                          '--name=vault_for_buildbot',
+                          '-p', '8200:8200', 'vault'])
         self.addCleanup(self.remove_container)
 
-        rv = subprocess.call(['docker', 'exec',
-                              '-e', 'VAULT_ADDR=http://127.0.0.1:8200/',
-                              'vault_for_buildbot',
-                              'vault', 'kv', 'put', 'secret/key', 'value=word'])
-        self.assertEqual(rv, 0)
```

Comment:
I'm not sure why the `docker exec` was failing but the `docker run` did not. Maybe the job didn't start fast enough since it runs with `-d`? Either way this guards the whole `setUp` with a skip if docker fails.